libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -1327,13 +1352,62 @@ void AP_BLHeli::update_telemetry(void)
     if (nbytes == telem_packet_size) {
         // we have a full packet ready to parse
         read_telemetry_packet();
+        last_telem_byte_read_us = 0;
     }
     if (now - last_telem_request_us >= telem_rate_us) {
+        // ask the next ESC for telemetry
         last_telem_esc = (last_telem_esc + 1) % num_motors;
         uint16_t mask = 1U << motor_map[last_telem_esc];
         hal.rcout->set_telem_request_mask(mask);
         last_telem_request_us = now;
     }
 }
 
-#endif // HAL_SUPPORT_RCOUT_SERIAL
+
+/*
+  send ESC telemetry messages over MAVLink
+ */
+void AP_BLHeli::send_esc_telemetry_mavlink(uint8_t mav_chan)
```

Comment:
Why not pass a `mavlink_channel_t`? (You have to cast to uint8_t to call this, then cast back to actually do the real message)

Reply:
because I don't want to bring the mavlink headers into AP_BLHeli.h